samples/subsys/mgmt/mcumgr/smp_svr/src/main.c

```diff
@@ -163,6 +168,24 @@ void main(void)
 	smp_bt_register();
 #endif
 
+#ifdef CONFIG_MCUMGR_SMP_UDP
+	printk("awaiting eth interface...\n");
+
+	struct net_if *iface = net_if_get_default();
+
+	while (!net_if_is_up(iface)) {
```

Comment:
This is a big no-no. We have infrastructure in zephyr to wait connectivity event and you should use that. See `samples/net/sockets/echo_client/src/echo-client.c` for details.